src/IterationTraits.php

```diff
@@ -299,4 +299,33 @@ public static function transpose(Iterator $iterator) {
                 });
         });
     }
-}
\ No newline at end of file
+
+    /**
+     * Reassembles a traversed sequence into its original shape
+     *
+     * @param Iterator $iterator
+     * @return Sequence
+     */
+    public static function reassemble(Iterator $iterator) {
+        return self::wrapFunctionIntoSequenceOnDemand(function() use ($iterator) {
+           return Sequence::make($iterator)
+               ->reduceToSequence(array(), function ($collection, $value, $path) {
+                   $keys = explode('.', $path);
```

Comment:
We have a getter/setter library that will inject the values into an array.
`use Revinate\GetterSetter as gs;`

`
return gs\set($collection, $path, $value);
`